overlord/devicestate/devicemgr.go

```diff
@@ -1543,6 +1543,56 @@ func (m *DeviceManager) ensurePostFactoryReset() error {
 	return os.Remove(factoryResetMarker)
 }
 
+func (m *DeviceManager) hasUserExpired(user *auth.UserState) bool {
```

Comment:
this isn't using anything from DeviceManager, it seems more an helper that could live in users.go or even become a method on UserState itself?

Reply:
I like the suggestion about integrating it into UserState itself, i'll do that!

Reply:
Fixed